File: src/lec08/Lec08Main.java
package lec08;

public class Lec08Main {

  public static void main(String[] args) {
  }

  public int max(int a, int b) {
    if (a > b) {
      return a;
    }
    return b;
  }

  public void repeat(String str, int num, boolean useNewLine) {
    for (int i = 1; i <= num; i++) {
      if (useNewLine) {
        System.out.println(str);
      } else {
        System.out.println(str);
      }
    }
  }

  public void repeat(String str, int num) { //메서드 오버로딩
    repeat(str, num, true);
  }

  public void repeat(String str) { //메서드 오버로딩
    repeat(str, 3, true);
  }

  public static void printAll(String... strs) { //가변인자
    for (String str : strs) {
      System.out.println(str);
    }
  }
}

File: src/lec08/Lec08Main.kt
package lec08

fun main() {
    repeat("Hello World")

    //가변인자
    printAll("A", "B", "C")

    val arr = arrayOf("A", "B", "C")
    printAll(*arr) //* : spread 연산자
}

fun max(a: Int, b: Int): Int {
    return if (a > b) {
        a
    }
    else b
}

fun max2(a: Int, b: Int): Int =
    if (a > b) {
        a
    } else {
        b
    }

fun max3(a: Int, b: Int) = if (a > b) a else b //반환 타입 생략

//default parameter
fun repeat(
    str: String,
    num: Int = 3,
    useNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (useNewLine) {
            println(str)
        } else {
            println(str)
        }
    }
}

//가변인자
fun printAll(vararg strs: String) {
    for (str in strs) {
        println(str)
    }
}